src/java.desktop/macosx/classes/com/apple/laf/AquaUtils.java

```diff
@@ -151,12 +149,14 @@ T get() {
     }
 
     abstract static class RecyclableSingleton<T> {
-        final T get() {
-            return AppContext.getSoftReferenceValue(this, () -> getInstance());
-        }
 
-        void reset() {
-            AppContext.getAppContext().remove(this);
+        T instance;
+
+        final T get() {
```

Comment:
It is used in a few places to cache the fonts/images/etc, so to be a "Recyclable" + "Singleton" it should store/return soft reference, and somehow handle the value types properly.